File: solutions/lua/eliuds-eggs/1/eliuds-eggs.lua
local EliudsEggs = {}

function EliudsEggs.egg_count(number)
    local num = number
    local count = 0
    while num > 0 do
      if(num % 2 == 1) then
        count = count + 1
      end
      num = num // 2
    end
  return count
end

return EliudsEggs

File: solutions/elixir/eliuds-eggs/1/lib/eliuds_eggs.ex
defmodule EliudsEggs do
  @doc """
  Given the number, count the number of eggs.
  """
  @spec egg_count(number :: integer()) :: non_neg_integer()
  def egg_count(number) when is_integer(number) and number >= 0 do
    cond do
      number == 0 -> 0
      Bitwise.band(number, 0b1) == 0b1 -> 1 + egg_count(Bitwise.bsr(number, 1))
      true -> egg_count(Bitwise.bsr(number, 1))
    end
  end
end
